src/Appwrite/Platform/Modules/Databases/Http/Databases/Collections/Documents/Create.php

```diff
@@ -350,6 +350,18 @@ public function action(string $databaseId, string $documentId, string $collectio
 
             // Assign a unique ID if needed, otherwise use the provided ID.
             $document['$id'] = $sourceId === 'unique()' ? ID::unique() : $sourceId;
+
+            // Allowing to add createdAt and updatedAt timestamps if server side(api key
+            if (!$isAPIKey && !$isPrivilegedUser) {
+                if (isset($document['$createdAt'])) {
+                    throw new Exception($this->getInvalidStructureException(), 'Attribute "$createdAt" is not allowed');
```

Comment:
After some discussion, let's make the message here a bit more verbose: `Attribute "$createdAt" can not be modified. Please use a server SDK with an API key to modify server attributes.`